code/modules/antagonists/eldritch_cult/eldritch_effects.dm

```diff
@@ -115,34 +115,77 @@
  */
 /datum/reality_smash_tracker
 	///list of tracked reality smashes
-	var/smashes = 0
+	var/list/smashes = list()
 	///List of mobs with ability to see the smashes
-	var/targets = 0
+	var/list/targets = list()
 
 /datum/reality_smash_tracker/Destroy(force, ...)
 	if(GLOB.reality_smash_track == src)
-		stack_trace("/datum/reality_smash_tracker was deleted. New heretics will no longer generate new influences")
+		stack_trace("/datum/reality_smash_tracker was deleted. Heretics may no longer access any influences. Fix it or call coder support")
+	QDEL_LIST(smashes)
+	targets.Cut()
 	return ..()
+/**
+ * Automatically fixes the target and smash network
+ *
+ * Fixes any bugs that are caused by late Generate() or exchanging clients
+ */
+/datum/reality_smash_tracker/proc/ReworkNetwork()
+	listclearnulls(smashes)
+	for(var/mind in targets)
+		if(isnull(mind))
+			stack_trace("A null somehow landed in a list of minds")
+			continue
+		for(var/X in smashes)
+			var/obj/effect/reality_smash/reality_smash = X
+			reality_smash.AddMind(mind)
 
 /**
  * Generates a set amount of reality smashes based on the N value
  *
  * Automatically creates more reality smashes
  */
-/datum/reality_smash_tracker/proc/Generate()
-	targets++
-	var/number = max(targets * (4-(targets-1)) - smashes,1)
+/datum/reality_smash_tracker/proc/Generate(mob/caller)
+	if(istype(caller))
+		targets += caller
+	var/targ_len = length(targets)
+	var/smash_len = length(smashes)
+	var/number = max(targ_len * (4-(targ_len-1)) - smash_len,1)
 
 	for(var/i in 0 to number)
-
 		var/turf/chosen_location = get_safe_random_station_turf()
 		//we also dont want them close to each other, at least 1 tile of seperation
 		var/obj/effect/reality_smash/what_if_i_have_one = locate() in range(1, chosen_location)
 		var/obj/effect/broken_illusion/what_if_i_had_one_but_got_used = locate() in range(1, chosen_location)
 		if(what_if_i_have_one || what_if_i_had_one_but_got_used) //we dont want to spawn
 			continue
 		new /obj/effect/reality_smash(chosen_location)
-		smashes++
+	ReworkNetwork()
+
+/**
+ * Adds a mind to the list of people that can see the reality smashes
+ *
+ * Use this whenever you want to add someone to the list
+ */
+/datum/reality_smash_tracker/proc/AddMind(datum/mind/M)
+	RegisterSignal(M.current,COMSIG_MOB_LOGIN,.proc/ReworkNetwork)
+	targets |= M
+	Generate()
+	for(var/X in smashes)
```

Comment:
`_reality_smash`

Reply:
I might be missing the point but changing `smashes` to `_reality_smash` will break because its iterating trough the list `smashes` in `GLOB.reality_smash_tracker`

Reply:
changing `X` to `_reality_smash`